src/databricks/labs/ucx/cli.py

```diff
@@ -268,11 +296,112 @@ def principal_prefix_access(w: WorkspaceClient, subscription_id: str | None = No
                 "or use the '--aws-profile=[profile-name]' parameter."
             )
             return None
-        return _aws_principal_prefix_access(w, aws_profile)
+        return func_aws(w, aws_profile)
     logger.error("This cmd is only supported for azure and aws workspaces")
     return None
 
 
+def _azure_setup_migration_principal(w: WorkspaceClient, subscription_id: str):
+    # issue: https://github.com/databrickslabs/ucx/issues/881
+    raise NotImplementedError
+
+
+def _aws_setup_migration_principal(w: WorkspaceClient, aws_profile: str):
+    if not shutil.which("aws"):
+        logger.error("Couldn't find AWS CLI in path. Please install the CLI from https://aws.amazon.com/cli/")
+        return
+
+    installation = Installation.current(w, 'ucx')
+    config = installation.load(WorkspaceConfig)
+
+    if not config.policy_id:
+        msg = "Cluster policy not found in UCX config"
+        logger.error(msg)
+        return
+
+    sql_backend = StatementExecutionBackend(w, config.warehouse_id)
+    location_crawler = ExternalLocations(w, sql_backend, config.inventory_database)
+    external_locations = location_crawler.snapshot()
+    s3_paths = {loc.location for loc in external_locations}
+
+    if len(s3_paths) == 0:
+        logger.info("No S3 paths to migrate found")
+        return
+
+    aws_permissions = AWSResourcePermissions.for_cli(w, sql_backend, aws_profile, config.inventory_database)
+    cluster_policy = _get_cluster_policy(w, config.policy_id)
+    iam_role_name_in_cluster_policy = _get_iam_role_from_cluster_policy(str(cluster_policy.definition))
+
+    iam_policy_name = f"UCX_MIGRATION_POLICY_{config.inventory_database}"
+    prompts = Prompts()
+    if iam_role_name_in_cluster_policy and aws_permissions.is_role_exists(iam_role_name_in_cluster_policy):
```

Comment:
this is too much logic for `cli.py`. move this into `AWSPermissions` method and pass it `Prompts` instance, so that it's unit testable. See `service_principal_migration.run(prompts)` as the example